ci.sh

```diff
@@ -137,7 +137,7 @@ pgrep -f target/debug/parsec >/dev/null
 if [ "$PROVIDER_NAME" = "all" ]; then
     echo "Execute all-providers tests"
     RUST_BACKTRACE=1 cargo test $TEST_FEATURES --manifest-path ./e2e_tests/Cargo.toml all_providers
-    RUST_BACKTRACE=1 cargo test $TEST_FEATURES --manifest-path ./e2e_tests/Cargo.toml config
+    RUST_BACKTRACE=1 cargo test $TEST_FEATURES --manifest-path ./e2e_tests/Cargo.toml config -- --test-threads=1
```

Comment:
Is testing in single-threaded mode needed even with the mutex we have now in PKCS11 provider?

Reply:
That was a mistake in the first place, not putting it like that. If you think about it, each of those tests changes the configuration of the service and expects it to be in a certain configuration while it's running. Multiple tests at the same time make no sense.

Reply:
oh right those are the config tests! Agree, then